src/features/projectsV2/ProjectsMap/index.tsx

```diff
@@ -0,0 +1,210 @@
+    selectedMode: props.isMobile ? props.selectedMode : undefined,
+    setSelectedMode: props.isMobile ? props.setSelectedMode : undefined,
+    isMobile: props.isMobile,
+    page: props.page,
+    mobileOS,
+  };
+
+  const onMouseMove = useCallback(
+    (e) => {
+      if (props.page !== 'project-details') return;
+      const hoveredPlantLocation = getPlantLocationInfo(
+        plantLocations,
+        mapRef,
+        e.point
+      );
+      if (
+        !hoveredPlantLocation ||
+        hoveredPlantLocation.hid === selectedPlantLocation?.hid
+      ) {
+        setHoveredPlantLocation(null);
+        return;
+      }
+      setHoveredPlantLocation(hoveredPlantLocation);
+    },
+    [plantLocations, props.page, selectedPlantLocation]
+  );
+
+  const onClick = useCallback(
+    (e) => {
+      if (props.page !== 'project-details') return;
+      const hasNoSites = singleProject?.sites?.length === 0;
+      const result = getPlantLocationInfo(plantLocations, mapRef, e.point);
+
+      const isSamePlantLocation =
+        result?.geometry.type === 'Point' &&
+        result.id === selectedPlantLocation?.id;
+      const isSingleTree =
+        selectedPlantLocation?.type === 'single-tree-registration';
+      const isMultiTree =
+        selectedPlantLocation?.type === 'multi-tree-registration';
+      // Clear sample plant location on clicking outside.
+      // Clicks on sample plant location will not propagate on the map
+      setSelectedSamplePlantLocation(null);
+      // Clear plant location info if clicked twice (single or multi tree) // point plant location
+      if (isSamePlantLocation && (isSingleTree || isMultiTree)) {
+        setSelectedPlantLocation(null);
+        setSelectedSite(hasNoSites ? null : 0);
+        return;
+      }
+
+      // Set selected plant location if a result is found
+      if (result) {
+        setSelectedSite(null);
+        setSelectedPlantLocation(result);
+      }
+    },
+    [plantLocations, props.page, selectedPlantLocation]
+  );
+
+  const singleProjectViewProps = {
+    mapRef,
+  };
+  const multipleProjectsViewProps = {
+    mapRef,
+    setViewState,
+    page: props.page,
+  };
+  const mapContainerClass = `${styles.mapContainer} ${
+    styles[mobileOS !== undefined ? mobileOS : '']
+  }`;
+  return (
+    <>
+      <MapControls {...mapControlProps} />
+      <div className={mapContainerClass}>
+        <Map
+          {...viewState}
+          {...mapState}
+          onMove={(e) => setViewState(e.viewState)}
+          onMouseMove={onMouseMove}
+          onMouseOut={() => setHoveredPlantLocation(null)}
+          onClick={onClick}
+          attributionControl={false}
+          ref={mapRef}
+          interactiveLayerIds={
+            singleProject !== null
+              ? ['plant-polygon-layer', 'point-layer']
+              : undefined
+          }
+        >
+          {shouldShowSingleProjectsView && (
+            <SingleProjectView {...singleProjectViewProps} />
```

Comment:
[] Address the TS warning

'SingleProjectView' cannot be used as a JSX component.
  Its return type 'Element | undefined' is not a valid JSX element.
    Type 'undefined' is not assignable to type 'Element | null'.

Reply:
Resolved in 6d8760b8b91e942f73b3dde3e7fafefeb05e2aaf